versioned_hdf5/tests/test_subchunk_map.py

```diff
@@ -1,136 +1,135 @@
-from collections import defaultdict
+from __future__ import annotations
+
+from typing import Any
 
 import hypothesis
-import ndindex
 import numpy as np
 import pytest
 from hypothesis import given
 from hypothesis import strategies as st
 from hypothesis.extra import numpy as stnp
-from numpy.testing import assert_equal
+from numpy.testing import assert_array_equal
 
 from ..subchunk_map import as_subchunk_map
 
+max_examples = 10_000
+
 
-def non_negative_step_slices(size):
-    start = st.one_of(st.integers(min_value=-size, max_value=size - 1), st.none())
-    stop = st.one_of(st.integers(min_value=-size, max_value=size), st.none())
+def non_negative_step_slices(size: int):
+    start = st.one_of(st.integers(-size - 1, size + 1), st.none())
+    stop = st.one_of(st.integers(-size - 1, size + 1), st.none())
```

Comment:
Since you're defining slices, the `max_value` can be anything and still work; but for testing purposes why is `max_value` not `size` rather than `size+1`?

Reply:
Because having a slice that exceeds the size of the array being sliced is an edge use case worth testing. Now, in practice, this edge case is negated by the call to `ndindex.ndindex.reduce()` which happens inside as_chunk_submap, but I don't want to assume it will always be there in this test.